topi/python/topi/x86/conv2d_transpose.py

```diff
@@ -16,50 +16,76 @@
 # under the License.
 # pylint: disable=invalid-name,unused-variable,unused-argument,no-member
 """Conv2D Transpose schedule on x86"""
-
 import tvm
 from tvm import autotvm
-from .. import generic, tag
-from ..nn.conv2d_transpose import conv2d_transpose_nchw, declaration_conv2d_transpose_impl
+from .. import generic
+from ..util import get_const_tuple, traverse_inline
+from ..nn import conv2d_transpose_nchw, dilate, pad, get_pad_tuple
+from . import conv2d_avx_1x1, conv2d_avx_common
+from .conv2d import _declaration_conv_impl, \
+    _create_tuning_space as _create_tuning_space_conv2d, \
+    _get_default_config as _get_default_config_conv2d
+
 
 @autotvm.register_topi_compute(conv2d_transpose_nchw, 'cpu', ['direct'])
-def _declaration_conv2d_transpose(cfg, data, kernel, strides, padding, out_dtype):
-    # TODO cfg is not used for now
-    return declaration_conv2d_transpose_impl(data, kernel, strides, padding, out_dtype)
+def _conv2d_transpose_nchw(cfg, data, kernel, strides, padding, out_dtype):
```

Comment:
Can we unify the following compute definition with the one [here](https://github.com/dmlc/tvm/blob/master/topi/python/topi/nn/conv2d_transpose.py#L56)?

Reply:
Now the declaration of covn2d_transpose in topi.nn and topi.x86 is unified using a conv2d_transpose_nchw_preprocess function.

CUDA still has its own declaration, because a separate dilation stage will increase shared memory usage and hurt the performance. To overcome this issue, the conv2d_transpose declaration in CUDA uses a super hacky way to inline dilation: https://github.com/dmlc/tvm/blob/master/topi/python/topi/cuda/conv2d_transpose_nchw.py#L104. I think it's fine to leave it unchanged.